benches/range_string-creation_functions.rs

```diff
@@ -0,0 +1,585 @@
+// benches/range_string-creation_functions.rs : evaluates performance of different range-string creation-function approaches
+
+#![allow(non_snake_case)]
+
+#![feature(ascii_char)]
+
+use shwild;
+
+use criterion::{
+    black_box,
+    criterion_group,
+    criterion_main,
+    // BenchmarkId,
+    Criterion,
+};
+
+
+mod constants {
+    #![allow(non_upper_case_globals)]
+    #![allow(unused)]
+
+    pub(crate) const EMPTY_STRING : &str = "";
+    pub(crate) const RANGE_STRING_abcd : &str = "abcd";
+    pub(crate) const RANGE_STRING_ALPHABET : &str = "abcdefghijklmnopqrstuvwxyz";
+    pub(crate) const RANGE_STRING_ALPHABET_MIXED_CASE : &str = "abcDefGHijklMNOPqrstUVwXyz";
+    pub(crate) const RANGE_STRING_DIGITS : &str = "0123456789";
+    pub(crate) const RANGE_STRING_MIXED_ASCII : &str = "!\"#$%&'()*+,-./3x0123456789:;<=>?4x@ABCDEFGHIJKLMNO5xPQRSTUVWXYZ[\\]^_6x`abcdefghijklmno7xpqrstuvwxyz{|}~";
+}
+
+
+mod utils {
+    #![allow(dead_code)]
+
+
+    pub(super) fn range_string_from_slice_X(
+        chars : &[char],
+        flags : i64,
+    ) -> String {
+        // range_string_from_slice_0(chars, flags)
+        // range_string_from_slice_1(chars, flags)
+        range_string_from_slice_2(chars, flags)
+        // range_string_from_slice_3(chars, flags)
+        // range_string_from_slice_4(chars, flags)
+        // range_string_from_slice_5(chars, flags)
+        // range_string_from_slice_6(chars, flags)
+        // range_string_from_slice_7(chars, flags)
+    }
+
+    pub(super) fn range_string_from_slice_0(
+        chars : &[char],
+        flags : i64,
+    ) -> String {
+        let mut chars = if 0 != (shwild::IGNORE_CASE & flags) {
+            let mut ci_chars = Vec::with_capacity(chars.len() * 2);
+
+            for c in chars {
+                ci_chars.push(c.to_ascii_uppercase());
+                ci_chars.push(c.to_ascii_lowercase());
+            }
+
+            ci_chars
+        } else {
+            chars.into()
+        };
+
+        chars.sort_unstable();
+
+        chars.dedup();
+
+        chars.into_iter().collect() // appears to be (marginally) faster for `Vec<>`, although it's very close
```

Comment:
This should be faster because Rust has specializations to try to reuse the buffer when calling `into_iter()`.